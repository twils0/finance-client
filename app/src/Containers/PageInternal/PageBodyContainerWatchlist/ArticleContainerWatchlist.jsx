import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestStatusTypes } from '../../../Constants/universalConstants';
import { statusNames } from '../../../Constants/dataConstantsWatchlist';
import requestSecurityData from '../../../Actions/dataThunkWatchlist/requestSecurityData';

import ArticleWatchlist from '../../../Components/PageInternal/PageBodyWatchlist/Article/ArticleWatchlist';

class ArticleContainerWatchlist extends React.Component {
  constructor(props) {
    super(props);

    const { securities } = this.props;
    const { current } = securities;

    if (current && securities[current]) {
      const currentSecurity = securities[current];

      const { handleSecurityData } = this.props;
      const { id, tickerCusip } = currentSecurity;

      console.log('test con sec data 1');

      handleSecurityData({ id, tickerCusip });
    }
  }

  componentWillReceiveProps(nextProps) {
    const { securities } = this.props;
    const newSecurities = nextProps.securities;

    if (securities.current !== newSecurities.current) {
      const { handleSecurityData } = this.props;
      const { current } = newSecurities;

      if (current && newSecurities[current]) {
        const currentSecurity = newSecurities[current];

        const { id, tickerCusip } = currentSecurity;

        handleSecurityData({ id, tickerCusip });
      }
    }
  }

  componentWillUnmount() {
    console.log('test art unmount');
  }

  render() {
    const {
      outsideWidthTotal,
      widthTable,
      widthBuffer,
      widthIndent,
      mediaWidth,
      menuHeight,
      articleMenuHeight,
      heightBuffer,
      heightIndent,
      status,
      securities,
    } = this.props;
    const statusLoading = status[statusNames.SECURITIES].status !== requestStatusTypes.SUCCESS
      || status[statusNames.GET_SECURITY_DATA].status === requestStatusTypes.LOADING;

    return (
      <ArticleWatchlist
        outsideWidthTotal={outsideWidthTotal}
        widthTable={widthTable}
        widthBuffer={widthBuffer}
        widthIndent={widthIndent}
        mediaWidth={mediaWidth}
        menuHeight={menuHeight}
        articleMenuHeight={articleMenuHeight}
        heightBuffer={heightBuffer}
        heightIndent={heightIndent}
        statusLoading={statusLoading}
        securities={securities}
      />
    );
  }
}

ArticleContainerWatchlist.propTypes = {
  outsideWidthTotal: PropTypes.number.isRequired,
  widthTable: PropTypes.number.isRequired,
  widthBuffer: PropTypes.number.isRequired,
  widthIndent: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  menuHeight: PropTypes.number.isRequired,
  articleMenuHeight: PropTypes.number.isRequired,
  heightBuffer: PropTypes.number.isRequired,
  heightIndent: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
  status: PropTypes.object.isRequired,
  handleSecurityData: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  outsideWidthTotal: state.ui.internal.watchlist.utilities.article.outsideWidthTotal,
  widthTable: state.ui.internal.watchlist.utilities.articleTable.width,
  widthBuffer: state.ui.internal.watchlist.utilities.article.widthBuffer,
  widthIndent: state.ui.internal.watchlist.utilities.article.widthIndent,
  mediaWidth: state.ui.internal.watchlist.utilities.pageBody.mediaWidth,
  menuHeight: state.ui.internal.utilities.menu.height,
  articleMenuHeight: state.ui.internal.watchlist.utilities.articleMenu.height,
  heightBuffer: state.ui.internal.watchlist.utilities.article.heightBuffer,
  heightIndent: state.ui.internal.watchlist.utilities.article.heightIndent,
  securities: state.data.watchlist.securities,
  status: state.data.watchlist.status,
});

const mapDispatchToProps = dispatch => ({
  handleSecurityData: payload => dispatch(requestSecurityData(payload)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ArticleContainerWatchlist);
