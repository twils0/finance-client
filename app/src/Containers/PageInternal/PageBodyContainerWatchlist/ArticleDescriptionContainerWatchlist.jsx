import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Paragraph from '../../../Components/Paragraph';

const ArticleDescriptionContainerWatchlist = (props) => {
  const { securities } = props;
  const { current } = securities;

  if (current && securities[current]) {
    const currentSecurity = securities[current];

    if (currentSecurity.data && currentSecurity.data.description) {
      const { articleDescriptionMinWidth } = props;
      const articleDescriptionMinWidthString = `${articleDescriptionMinWidth}px`;
      const { description } = currentSecurity.data;

      return (
        <Paragraph
          flex="1 1"
          alignSelf="flex-start"
          margin="0"
          width="100%"
          minWidth={articleDescriptionMinWidthString}
        >
          {description}
        </Paragraph>
      );
    }
  }
  return null;
};

ArticleDescriptionContainerWatchlist.propTypes = {
  articleDescriptionMinWidth: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  articleDescriptionMinWidth: state.ui.internal.watchlist.utilities.articleDescription.minWidth,
  securities: state.data.watchlist.securities,
});

export default connect(mapStateToProps)(ArticleDescriptionContainerWatchlist);
