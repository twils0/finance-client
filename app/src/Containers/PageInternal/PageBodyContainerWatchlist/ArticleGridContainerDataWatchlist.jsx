import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { statusNames } from '../../../Constants/dataConstantsWatchlist';

import ArticleGridDataWatchlist from '../../../Components/PageInternal/PageBodyWatchlist/Article/ArticleGridDataWatchlist';

const ArticleGridContainerDataWatchlist = (props) => {
  const {
    securities, articleGridWidthIndent, articleGridHeightBuffer, articleGridHeightIndent,
  } = props;

  const { current } = securities;

  if (current && securities[current]) {
    const currentSecurity = securities[current];

    if (currentSecurity.data && currentSecurity.data.price) {
      const dataPrice = currentSecurity.data.price;

      return (
        <ArticleGridDataWatchlist
          widthIndent={articleGridWidthIndent}
          heightBuffer={articleGridHeightBuffer}
          heightIndent={articleGridHeightIndent}
          dataPrice={dataPrice}
        />
      );
    }
  }
  return null;
};

ArticleGridContainerDataWatchlist.propTypes = {
  articleGridWidthIndent: PropTypes.number.isRequired,
  articleGridHeightBuffer: PropTypes.number.isRequired,
  articleGridHeightIndent: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  articleGridWidthIndent: state.ui.internal.watchlist.utilities.articleGrid.widthIndent,
  articleGridHeightBuffer: state.ui.internal.watchlist.utilities.articleGrid.heightBuffer,
  articleGridHeightIndent: state.ui.internal.watchlist.utilities.articleGrid.heightIndent,
  statusSecurityData: state.data.watchlist.status[statusNames.GET_SECURITY_DATA].status,
  securities: state.data.watchlist.securities,
});

export default connect(mapStateToProps)(ArticleGridContainerDataWatchlist);
