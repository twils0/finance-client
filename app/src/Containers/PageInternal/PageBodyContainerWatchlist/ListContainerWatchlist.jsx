import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { deepEqual } from 'fast-equals';

import { pathNames } from '../../../Constants/universalConstants';
import { setSecuritiesList } from '../../../Actions/dataActionsWatchlist';
import requestUpdateSecurities from '../../../Actions/dataThunkWatchlist/requestUpdateSecurities';

import ListWatchlist from '../../../Components/PageInternal/PageBodyWatchlist/ListWatchlist';

class ListContainerWatchlist extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      order: props.securities.list.map((security, index) => index),
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setOrder(nextProps.securities.list);
  }

  setOrder = (list) => {
    if (list.length !== this.state.order.length) {
      this.resetOrder(list);
    }
  };

  resetOrder = (list) => {
    const order = list.map((security, index) => index);
    this.setState({ order });
  };

  updateLocalStateOrder = (order) => {
    this.setState({ order });
  };

  updateSecurities = () => {
    const { securities, handleSecuritiesList, handleUpdateSecurities } = this.props;
    const { order } = this.state;
    const list = [];

    order.forEach((index) => {
      list.push(securities.list[index]);
    });

    if (!deepEqual(list, securities.list)) {
      this.resetOrder(list);
      handleSecuritiesList({ list });
      handleUpdateSecurities();
    }
  };

  handleItemClick = (clickEvent) => {
    const { securities } = this.props;
    const securityId = clickEvent.target.id;

    if (securities.current !== securityId && Object.keys(securities).indexOf(securityId) > -1) {
      this.props.history.push(`${pathNames.WATCHLIST}/${securityId}`);
    }
  };

  render() {
    const {
      springSettings,
      width,
      widthBufferLeft,
      widthBufferScrollBar,
      widthBufferRight,
      heightBuffer,
      responsiveHeight,
      rowWidthIndent,
      rowHeight,
      rowHeightIndent,
      rowHeightBufferBottom,
      burgerMenuIconWidth,
      burgerMenuIconWidthIndent,
      burgerMenuIconHeight,
      securities,
    } = this.props;
    const { order } = this.state;

    return (
      <ListWatchlist
        springSettings={springSettings}
        width={width}
        widthBufferLeft={widthBufferLeft}
        widthBufferScrollBar={widthBufferScrollBar}
        widthBufferRight={widthBufferRight}
        heightBuffer={heightBuffer}
        responsiveHeight={responsiveHeight}
        rowWidthIndent={rowWidthIndent}
        rowHeight={rowHeight}
        rowHeightIndent={rowHeightIndent}
        rowHeightBufferBottom={rowHeightBufferBottom}
        burgerMenuIconWidth={burgerMenuIconWidth}
        burgerMenuIconWidthIndent={burgerMenuIconWidthIndent}
        burgerMenuIconHeight={burgerMenuIconHeight}
        securities={securities}
        order={order}
        updateLocalStateOrder={this.updateLocalStateOrder}
        updateSecurities={this.updateSecurities}
        handleClick={this.handleItemClick}
      />
    );
  }
}

ListContainerWatchlist.propTypes = {
  history: PropTypes.object.isRequired,
  springSettings: PropTypes.shape({ stiffness: PropTypes.number, damping: PropTypes.number })
    .isRequired,
  width: PropTypes.number.isRequired,
  widthBufferLeft: PropTypes.number.isRequired,
  widthBufferScrollBar: PropTypes.number.isRequired,
  widthBufferRight: PropTypes.number.isRequired,
  heightBuffer: PropTypes.number.isRequired,
  responsiveHeight: PropTypes.number.isRequired,
  rowWidthIndent: PropTypes.number.isRequired,
  rowHeight: PropTypes.number.isRequired,
  rowHeightIndent: PropTypes.number.isRequired,
  rowHeightBufferBottom: PropTypes.number.isRequired,
  burgerMenuIconWidth: PropTypes.number.isRequired,
  burgerMenuIconWidthIndent: PropTypes.number.isRequired,
  burgerMenuIconHeight: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
  handleSecuritiesList: PropTypes.func.isRequired,
  handleUpdateSecurities: PropTypes.func.isRequired,
};

ListContainerWatchlist.defaultProps = {};

const mapStateToProps = (state, ownProps) => ({
  history: ownProps.history,
  springSettings: state.ui.internal.watchlist.utilities.list.springSettings,
  width: state.ui.internal.watchlist.utilities.list.width,
  widthBufferLeft: state.ui.internal.watchlist.utilities.list.widthBufferLeft,
  widthBufferScrollBar: state.ui.internal.watchlist.utilities.list.widthBufferScrollBar,
  widthBufferRight: state.ui.internal.watchlist.utilities.list.widthBufferRight,
  heightBuffer: state.ui.internal.watchlist.utilities.list.heightBuffer,
  responsiveHeight: state.ui.internal.watchlist.utilities.list.responsiveHeight,
  rowWidthIndent: state.ui.internal.watchlist.utilities.list.rowWidthIndent,
  rowHeight: state.ui.internal.watchlist.utilities.list.rowHeight,
  rowHeightIndent: state.ui.internal.watchlist.utilities.list.rowHeightIndent,
  rowHeightBufferBottom: state.ui.internal.watchlist.utilities.list.rowHeightBufferBottom,
  burgerMenuIconWidth: state.ui.internal.watchlist.utilities.list.burgerMenuIconWidth,
  burgerMenuIconWidthIndent: state.ui.internal.watchlist.utilities.list.burgerMenuIconWidthIndent,
  burgerMenuIconHeight: state.ui.internal.watchlist.utilities.list.burgerMenuIconHeight,
  securities: state.data.watchlist.securities,
});

const mapDispatchToProps = dispatch => ({
  handleSecuritiesList: payload => dispatch(setSecuritiesList(payload)),
  handleUpdateSecurities: payload => dispatch(requestUpdateSecurities(payload)),
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(ListContainerWatchlist),
);
