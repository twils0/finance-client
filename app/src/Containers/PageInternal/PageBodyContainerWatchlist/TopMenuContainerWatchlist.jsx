import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestStatusTypes } from '../../../Constants/universalConstants';
import { statusNames } from '../../../Constants/dataConstantsWatchlist';
import requestUpdateSecurities from '../../../Actions/dataThunkWatchlist/requestUpdateSecurities';
import deleteSecurity from '../../../Actions/dataThunkWatchlist/deleteSecurity';

import TopMenu from '../../../Components/TopMenu';
import FlexRow from '../../../Components/Flex/FlexRow';
import FlexColumn from '../../../Components/Flex/FlexColumn';
import Header2 from '../../../Components/Header2';
import ButtonUnsubscribe from '../../../Components/PageInternal/PageBodyWatchlist/Article/ButtonUnsubscribe';

class TopMenuContainerWatchlist extends React.Component {
  handleUnsubscribeButtonClick = (clickEvent) => {
    clickEvent.preventDefault();
    const { securities } = this.props;
    const { current } = securities;

    if (current && securities[current]) {
      const { handleDeleteSecurity, handleUpdateSecurities } = this.props;

      handleDeleteSecurity({ id: current });

      handleUpdateSecurities();
    }
  };

  render() {
    const {
      menuSecurityWidthBuffer,
      mediaWidth,
      menuSecurityHeightBuffer,
      securities,
    } = this.props;
    const menuSecurityWidthBufferHalfString = `${menuSecurityWidthBuffer / 2}px`;
    const mediaWidthString = `${mediaWidth}px`;
    const menuSecurityHeightBufferHalfString = `${menuSecurityHeightBuffer / 2}px`;
    let currentSecurity = null;
    let categoryExchangeTickerCusip = null;

    if (securities.current && securities[securities.current]) {
      currentSecurity = securities[securities.current];

      if (currentSecurity && currentSecurity.exchange) {
        categoryExchangeTickerCusip = `${currentSecurity.category} (${currentSecurity.exchange}: ${
          currentSecurity.tickerCusip
        })`;
      } else {
        categoryExchangeTickerCusip = `${currentSecurity.category} (${
          currentSecurity.tickerCusip
        })`;
      }
    }

    if (currentSecurity) {
      const { status } = this.props;
      const statusLoading = status[statusNames.SECURITIES].status !== requestStatusTypes.SUCCESS
        || status[statusNames.GET_SECURITY_DATA].status === requestStatusTypes.LOADING;

      return (
        <TopMenu
          alignItems="flex-end"
          justifyContent="center"
          responsivePadding={`0 0 ${menuSecurityHeightBufferHalfString} 0`}
          width="100%"
          mediaWidth={mediaWidthString}
          borderWidth="0 0 1px 0"
        >
          <FlexRow
            flex="1 1 auto"
            justifyContent="flex-start"
            padding={`0 ${menuSecurityWidthBufferHalfString}
            ${menuSecurityHeightBufferHalfString} 0`}
            width="100%"
            mediaWidth={mediaWidthString}
            responsivePadding={`${menuSecurityHeightBufferHalfString} 0`}
          >
            <FlexColumn
              width="100%"
              alignItems="flex-start"
              mediaWidth={mediaWidthString}
              responsiveAlignSecurities="center"
            >
              <Header2
                themeColor="secondary"
                colorType="default"
                fontSize="24px"
                fontFamily="Titillium Web"
                margin="0"
                mediaWidth={mediaWidthString}
                responsiveTextAlign="center"
                responsivePadding={`0 0 ${menuSecurityHeightBufferHalfString} 0`}
              >
                {currentSecurity && currentSecurity.name}
              </Header2>
              <Header2
                colorType="default"
                fontSize="17px"
                fontFamily="Titillium Web"
                margin="0"
                padding={`${menuSecurityHeightBufferHalfString} 0 0 0`}
                mediaWidth={mediaWidthString}
                responsiveTextAlign="center"
              >
                {categoryExchangeTickerCusip}
              </Header2>
            </FlexColumn>
          </FlexRow>
          <FlexRow
            flex="0 0 auto"
            padding={`${menuSecurityHeightBufferHalfString} 0
          ${menuSecurityHeightBufferHalfString} ${menuSecurityWidthBufferHalfString}`}
            mediaWidth={mediaWidthString}
            responsivePadding={`${menuSecurityHeightBufferHalfString} 0`}
          >
            <ButtonUnsubscribe
              statusLoading={statusLoading}
              handleClick={this.handleUnsubscribeButtonClick}
            />
          </FlexRow>
        </TopMenu>
      );
    }
    return null;
  }
}

TopMenuContainerWatchlist.propTypes = {
  menuSecurityWidthBuffer: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  menuSecurityHeightBuffer: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
  status: PropTypes.object.isRequired,
  handleDeleteSecurity: PropTypes.func.isRequired,
  handleUpdateSecurities: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  menuSecurityWidthBuffer: state.ui.internal.watchlist.utilities.articleMenuSecurity.widthBuffer,
  mediaWidth: state.ui.internal.watchlist.utilities.pageBody.mediaWidth,
  menuSecurityHeightBuffer: state.ui.internal.watchlist.utilities.articleMenuSecurity.heightBuffer,
  securities: state.data.watchlist.securities,
  status: state.data.watchlist.status,
});

const mapDispatchToProps = dispatch => ({
  handleDeleteSecurity: payload => dispatch(deleteSecurity(payload)),
  handleUpdateSecurities: () => dispatch(requestUpdateSecurities()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TopMenuContainerWatchlist);
