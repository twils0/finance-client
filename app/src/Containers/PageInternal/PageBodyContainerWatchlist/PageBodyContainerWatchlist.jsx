import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BeatLoader } from 'halogenium';

import theme from '../../../themes';

import { requestStatusTypes } from '../../../Constants/universalConstants';
import { statusNames } from '../../../Constants/dataConstantsWatchlist';

import PageBody from '../../../Components/PageBody';
import ListContainerWatchlist from './ListContainerWatchlist';
import FlexColumn from '../../../Components/Flex/FlexColumn';
import ArticleContainerWatchlist from './ArticleContainerWatchlist';

const PageBodyContainerWatchlist = (props) => {
  const {
    width, mediaWidth, heightBuffer, menuHeight, menuHeightBuffer, statusSecurities,
  } = props;
  const widthHalfString = `${width / 2}px`;
  const mediaWidthString = `${mediaWidth}px`;
  const heightString = `calc(100% - ${heightBuffer}px)`;
  const heightBufferHalfString = `${heightBuffer / 2}px`;
  const menuHeightString = `${menuHeight + menuHeightBuffer}px`;

  if (statusSecurities !== requestStatusTypes.SUCCESS) {
    return (
      <FlexColumn width="100%" height={`calc(100% - ${menuHeightString})`}>
        <BeatLoader color={theme.color.secondary.default} />
      </FlexColumn>
    );
  }

  console.log('test status sec', statusSecurities);

  return (
    <PageBody mediaWidth={mediaWidthString} height={`calc(100% - ${menuHeightString})`}>
      <ListContainerWatchlist />
      <FlexColumn
        themeColor="secondary"
        borderColorType="default"
        separator
        mediaWidth={mediaWidthString}
        height={heightString}
        margin={`${heightBufferHalfString} 0`}
        borderWidth={`0 ${widthHalfString}`}
        borderRadius="4px"
      />
      <ArticleContainerWatchlist />
    </PageBody>
  );
};

PageBodyContainerWatchlist.propTypes = {
  width: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  heightBuffer: PropTypes.number.isRequired,
  menuHeight: PropTypes.number.isRequired,
  menuHeightBuffer: PropTypes.number.isRequired,
  statusSecurities: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  width: state.ui.internal.watchlist.utilities.separatorLine.width,
  mediaWidth: state.ui.internal.watchlist.utilities.pageBody.mediaWidth,
  heightBuffer: state.ui.internal.watchlist.utilities.separatorLine.heightBuffer,
  menuHeight: state.ui.internal.utilities.menu.height,
  menuHeightBuffer: state.ui.internal.utilities.menu.heightBuffer,
  statusSecurities: state.data.watchlist.status[statusNames.SECURITIES].status,
});

export default connect(mapStateToProps)(PageBodyContainerWatchlist);
