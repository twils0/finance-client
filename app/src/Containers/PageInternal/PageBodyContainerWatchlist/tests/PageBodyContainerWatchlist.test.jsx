import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateAWS } from '../../../../Reducers/dataReducersAWS';
import { initialState as initialStateWatchlist } from '../../../../Reducers/dataReducersWatchlist';
import { initialState as initialStateUIInternal } from '../../../../Reducers/uiReducersInternal';
import { initialState as initialStateUIWatchlist } from '../../../../Reducers/uiReducersWatchlist';
import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsWatchlist';

import PageBodyContainerWatchlist from '../PageBodyContainerWatchlist';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const shallowComponent = store => shallow(<PageBodyContainerWatchlist />, {
  context: {
    store,
  },
}).dive();

const testId0 = 'testId0';
const testId1 = 'testId1';
const testId2 = 'testId2';

const securities = {
  current: testId0,
  list: [testId0, testId1, testId2],
  [testId0]: {
    id: testId0,
    name: 'testName0',
    category: 'testCategory0',
    exchange: 'testExchange0',
    tickerCusip: 'testTickerCusip0',
    data: {
      description: 'testDescription0',
    },
  },
  [testId1]: {
    id: testId1,
    name: 'testName1',
    category: 'testCategory1',
    exchange: 'testExchange1',
    tickerCusip: 'testTickerCusip1',
    data: {
      description: 'testDescription1',
    },
  },
  [testId2]: {
    id: testId2,
    name: 'testName2',
    category: 'testCategory2',
    exchange: 'testExchange2',
    tickerCusip: 'testTickerCusip2',
    data: {
      description: 'testDescription2',
    },
  },
};

describe('Containers', () => {
  describe('PageInternal', () => {
    describe('PageBodyContainerWatchlist', () => {
      it('shallow renders correctly', async () => {
        const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
        const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

        stateBeforeAWS.status = requestStatusTypes.SUCCESS;

        stateBeforeWatchlist.securities = securities;

        const store = mockStore({
          data: {
            aws: stateBeforeAWS,
            watchlist: stateBeforeWatchlist,
          },
          ui: {
            internal: {
              ...initialStateUIInternal,
              watchlist: initialStateUIWatchlist,
            },
          },
        });

        const expectedActions = [];

        const wrapper = shallowComponent(store);

        const actions = store.getActions();

        expect(actions).toEqual(expectedActions);
        expect(wrapper).toMatchSnapshot();
      });

      it('shallow renders correctly, with loading indicator', async () => {
        const store = mockStore({
          data: {
            aws: initialStateAWS,
            watchlist: initialStateWatchlist,
          },
          ui: {
            internal: {
              ...initialStateUIInternal,
              watchlist: initialStateUIWatchlist,
            },
          },
        });

        const expectedActions = [];

        const wrapper = shallowComponent(store);

        const actions = store.getActions();

        expect(actions).toEqual(expectedActions);
        expect(wrapper).toMatchSnapshot();
      });

      it('shallow renders correctly, statusSecurities equals succcess', async () => {
        const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

        stateBeforeWatchlist.securities = securities;
        stateBeforeWatchlist.status[statusNames.SECURITIES].status = requestStatusTypes.SUCCESS;

        const store = mockStore({
          data: {
            aws: initialStateAWS,
            watchlist: stateBeforeWatchlist,
          },
          ui: {
            internal: {
              ...initialStateUIInternal,
              watchlist: initialStateUIWatchlist,
            },
          },
        });

        const expectedActions = [];

        const wrapper = shallowComponent(store);

        const actions = store.getActions();

        expect(actions).toEqual(expectedActions);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
