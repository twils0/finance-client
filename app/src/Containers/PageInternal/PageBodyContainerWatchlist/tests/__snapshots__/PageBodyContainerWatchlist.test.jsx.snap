// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageInternal PageBodyContainerWatchlist shallow renders correctly 1`] = `
<FlexColumn
  alignItems={null}
  alignSelf={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexWrap={null}
  height="calc(100% - 103px)"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  position={null}
  responsiveAlignSecurities={null}
  responsiveJustifyContent={null}
  responsiveMargin={null}
  responsivePadding={null}
  responsivePosition={null}
  responsiveWidth={null}
  separator={false}
  themeColor="primary"
  width="100%"
>
  <BeatLoader
    color="#7367e6"
    loading={true}
    margin="2px"
    size="15px"
  />
</FlexColumn>
`;

exports[`Containers PageInternal PageBodyContainerWatchlist shallow renders correctly, statusSecurities equals succcess 1`] = `
<PageBody
  alignItems={null}
  background={null}
  flex={null}
  flexDirection={null}
  height="calc(100% - 103px)"
  justifyContent={null}
  margin={null}
  mediaWidth="925px"
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  width={null}
>
  <withRouter(Connect(ListContainerWatchlist)) />
  <FlexColumn
    alignItems={null}
    alignSelf={null}
    borderColorType="default"
    borderRadius="4px"
    borderStyle={null}
    borderWidth="0 2px"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height="calc(100% - 40px)"
    justifyContent={null}
    margin="20px 0"
    mediaWidth="925px"
    minHeight={null}
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    responsiveWidth={null}
    separator={true}
    themeColor="secondary"
    width={null}
  />
  <Connect(ArticleContainerWatchlist) />
</PageBody>
`;

exports[`Containers PageInternal PageBodyContainerWatchlist shallow renders correctly, with componentWillUpdate 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageBodyContainerWatchlist
    heightBuffer={40}
    mediaWidth={925}
    menuHeight={73}
    menuHeightBuffer={30}
    statusItems="IDLE"
    width={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": null,
      "alignSelf": null,
      "borderColorType": "border",
      "borderRadius": null,
      "borderStyle": null,
      "borderWidth": null,
      "boxSizing": null,
      "children": <BeatLoader
        color="#509236"
        loading={true}
        margin="2px"
        size="15px"
      />,
      "color": Object {},
      "display": null,
      "flex": null,
      "flexWrap": null,
      "height": "calc(100% - 103px)",
      "justifyContent": null,
      "margin": null,
      "mediaWidth": null,
      "minHeight": null,
      "minWidth": null,
      "overflowX": null,
      "overflowY": null,
      "padding": null,
      "position": null,
      "responsiveAlignItems": null,
      "responsiveJustifyContent": null,
      "responsiveMargin": null,
      "responsivePadding": null,
      "responsivePosition": null,
      "responsiveWidth": null,
      "separator": false,
      "themeColor": "primary",
      "width": "100%",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#509236",
        "loading": true,
        "margin": "2px",
        "size": "15px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": null,
        "alignSelf": null,
        "borderColorType": "border",
        "borderRadius": null,
        "borderStyle": null,
        "borderWidth": null,
        "boxSizing": null,
        "children": <BeatLoader
          color="#509236"
          loading={true}
          margin="2px"
          size="15px"
        />,
        "color": Object {},
        "display": null,
        "flex": null,
        "flexWrap": null,
        "height": "calc(100% - 103px)",
        "justifyContent": null,
        "margin": null,
        "mediaWidth": null,
        "minHeight": null,
        "minWidth": null,
        "overflowX": null,
        "overflowY": null,
        "padding": null,
        "position": null,
        "responsiveAlignItems": null,
        "responsiveJustifyContent": null,
        "responsiveMargin": null,
        "responsivePadding": null,
        "responsivePosition": null,
        "responsiveWidth": null,
        "separator": false,
        "themeColor": "primary",
        "width": "100%",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#509236",
          "loading": true,
          "margin": "2px",
          "size": "15px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;

exports[`Containers PageInternal PageBodyContainerWatchlist shallow renders correctly, with loading indicator 1`] = `
<FlexColumn
  alignItems={null}
  alignSelf={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexWrap={null}
  height="calc(100% - 103px)"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  position={null}
  responsiveAlignSecurities={null}
  responsiveJustifyContent={null}
  responsiveMargin={null}
  responsivePadding={null}
  responsivePosition={null}
  responsiveWidth={null}
  separator={false}
  themeColor="primary"
  width="100%"
>
  <BeatLoader
    color="#7367e6"
    loading={true}
    margin="2px"
    size="15px"
  />
</FlexColumn>
`;
